bluecellulab/simulation/simulation.py

```diff
@@ -50,26 +51,29 @@ def init_progress_callback(self):
 
     def progress_callback(self):
         """Callback function for the progress bar."""
-        if self.progress > 0:
-            sys.stdout.write("\x1b[3F")
-
-        self.progress += 1
-        self.progress_closed = not self.progress_closed
-        if self.progress_closed:
-            sys.stdout.write(" %s%s%s \n" % (" " * (
-                self.progress - 1), " ", " " * (100 - self.progress)))
-            sys.stdout.write("[%s%s%s]\n" % ("#" * (
-                self.progress - 1), "-", "." * (100 - self.progress)))
-            sys.stdout.write(" %s%s%s \n" % (" " * (
-                self.progress - 1), " ", " " * (100 - self.progress)))
+        if self.custom_progress_function is None:
+            if self.progress > 0:
+                sys.stdout.write("\x1b[3F")
+
+            self.progress += 1
+            self.progress_closed = not self.progress_closed
+            if self.progress_closed:
+                sys.stdout.write(" %s%s%s \n" % (" " * (
+                    self.progress - 1), " ", " " * (100 - self.progress)))
+                sys.stdout.write("[%s%s%s]\n" % ("#" * (
+                    self.progress - 1), "-", "." * (100 - self.progress)))
+                sys.stdout.write(" %s%s%s \n" % (" " * (
+                    self.progress - 1), " ", " " * (100 - self.progress)))
+            else:
+                sys.stdout.write(" %s%s%s \n" % (" " * (
+                    self.progress - 1), "/", " " * (100 - self.progress)))
+                sys.stdout.write("[%s%s%s]\n" % ("#" * (
+                    self.progress - 1), ">", "." * (100 - self.progress)))
+                sys.stdout.write(" %s%s%s \n" % (" " * (
+                    self.progress - 1), "\\", " " * (100 - self.progress)))
+            sys.stdout.flush()
         else:
-            sys.stdout.write(" %s%s%s \n" % (" " * (
-                self.progress - 1), "/", " " * (100 - self.progress)))
-            sys.stdout.write("[%s%s%s]\n" % ("#" * (
-                self.progress - 1), ">", "." * (100 - self.progress)))
-            sys.stdout.write(" %s%s%s \n" % (" " * (
-                self.progress - 1), "\\", " " * (100 - self.progress)))
-        sys.stdout.flush()
+            self.custom_progress_function()
 
         neuron.h.cvode.event(
             neuron.h.t + self.progress_dt, self.progress_callback)
```

Comment:
what if the cvode is disabled 
for `synaptome simulation,` we have to disable the `cvode`,  so i don't know if the event will be triggered hence the progress_callback will work ?

Reply:
I have tested with cvode disabled, and the callback function is still called.

Reply:
okey thanks